src/java/org/apache/cassandra/db/BufferDecoratedKey.java

```diff
@@ -39,6 +39,12 @@ public ByteBuffer getKey()
         return key;
     }
 
+    @Override
+    public int getKeyLength()
+    {
+        return key.remaining();
```

Comment:
I feel uneasy about expressing immutable concept of key length with perfectly mutable byte buffer position

Reply:
The buffer is treated as an immutable object here, as well as in too many other places in the codebase -- this is something this ticket cannot address, and if we do want to fix it, that needs to be done in the OSS codebase first (it would be a very sizeable change that would be hell to rebase).